src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java

```diff
@@ -3332,7 +3333,7 @@ public static void solve(ExecutionContext ec, GPUContext gCtx, String instName,
 	@SuppressWarnings("unused")
 	private static void debugPrintMatrix(Pointer in, int rlen, int clen){
 		double[] data = new double[rlen * clen];
-		cudaMemcpy(Pointer.to(data), in, rlen*clen*Sizeof.DOUBLE, cudaMemcpyDeviceToHost);
+		cudaMemcpy(Pointer.to(data), in, rlen*clen*handle.sizeOfDatatype(), cudaMemcpyDeviceToHost);
```

Comment:
I am sure these won't work right off the bat. 
You'd first need to cast them to a float array on the host machine, then copy them to the GPU.
(Also I understand this is a WIP, and there is more work to be done. In light of that, please ignore this comment if you had already planned to do what I suggested).